### ISSUE TITLE
Precious should work when run from subdirectories

### ISSUE THREAD

**Issue by oschwald** [houseabsolute/precious#29]
Right now, `precious` appears to give a `No such file or directory (os error 2)` error when run from a subdirectory.

For instance, given this `.precious.toml`:

```
[commands.test]
type = "tidy"
cmd = [ "golines" ]
ok_exit_codes = 0
include = [ "**/*" ]
```

and an empty file at `dir/test.go`. I get this output when running `precious tidy dir/test.go`:

```
💍 Tidying paths passed on the command line (recursively)
✨ Unchanged by test: dir/test.go
```

but if I do `cd dir && precious tidy test.go`, I get:

```
💍 Tidying paths passed on the command line (recursively)
💥 error test: test.go
Error when tidying files:
  ▶ test.go [commands.test]
    No such file or directory (os error 2)
```

The "No such file or directory" error appears to come from `precious` itself.

**Comment by autarch** [comment]
Would you be up for trying out the code in `in-dir` branch? It's kind of a big change so I'd like some independent testing.

**Comment by oschwald** [comment]
Thanks for making a branch.

I did a bit of testing of this. When using the `--git` flag from a subdirectory, I see similar errors as before.

When I pass a file by name, the `include` directives don't seem to be parsed correctly. For instance, if I have a linter set up with an `include` of `go/**/*.go` and I am in `go/foo` and run `precious lint main.go`, the linter is not run.

**Comment by autarch** [comment]
Yeah, that's not surprising. This was a hacky attempt at a fix. I think I need to rethink all the path handling across the codebase in order to get this working properly. I don't think it's consistent about how it handles relative vs absolute paths.

**Comment by oschwald** [comment]
Although I am not familiar with the code, would a simpler approach be to normalize any paths passed as arguments to either the absolute path or relative to the Precious root directory and then change the current working directory for the process to the root directory?

**Comment by autarch** [comment]
> Although I am not familiar with the code, would a simpler approach be to normalize any paths passed as arguments to either the absolute path or relative to the Precious root directory and then change the current working directory for the process to the root directory?

I think something like that would work, though for commands where `chdir` is true it has to just pass the filename.

**Comment by autarch** [comment]
Alright, I have a take 2 of this work. I added some integration tests to check path handling and I think I've got it working the way it should now.

I've changed up the repo structure a bit, so to install this new version you'll need to run `cargo install --path ./precious`.

If you could test this again I'd appreciate it.

**Comment by oschwald** [comment]
So far it looks good to me. I spent a bit of time testing various scenarios and I didn't find any issues with it. Thanks!

**Comment by autarch** [comment]
Great! Thanks for doing some manual testing on this. I tried to add more integration tests to cover various cases but I feel like it's easy to miss corner cases.

**Comment by autarch** [comment]
This should be fixed in v0.2.1.
